<!DOCTYPE html>
<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      novel extremophiles: cladogram construction &middot; jmicrobe
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link href='https://fonts.googleapis.com/css?family=Pompiere' rel='stylesheet' type='text/css'>

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-144-precomposed.png">
                                 <link rel="shortcut icon" href="/public/favicon.ico">

  <script src="https://use.fontawesome.com/b4fd5cd984.js"></script>


  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body>

    <div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      <h1>
        <center><img src="/public/wbunny.png" style="width: 45%;"></center>
        <a href="/">
          jmicrobe
        </a>
      </h1>
      <p class="lead">Microbial ecologist, bioinformatician, mad scientist</p>
    </div>

    <nav class="sidebar-nav">

      

      
      
        
          
        
      
        
          
            <a class="sidebar-nav-item" href="/about/">about me</a>
          
        
      
        
      
        
          
            <a class="sidebar-nav-item" href="/cv/">curriculum vitae</a>
          
        
      
        
          
        
      
        
          
            <a class="sidebar-nav-item" href="/portfolio/">portfolio</a>
          
        
      
        
          
            <a class="sidebar-nav-item" href="/posts/">posts</a>
          
        
      
        
          
            <a class="sidebar-nav-item" href="/tutorials/">tutorials</a>
          
        
      
        
      
      <p><center>
        <a href="https://github.com/jmicrobe"><i class="fa fa-github fa-lg" aria-hidden="true"></i></a>
        <a href="https://www.linkedin.com/in/hardwickejessica"><i class="fa fa-linkedin-square fa-lg" aria-hidden="true"></i></a>
        <a href="https://twitter.com/jessica_microbe"><i class="fa fa-twitter-square fa-lg" aria-hidden="true"></i></a>
        <a href="mailto:jessica@jmicrobe.me"><i class="fa fa-envelope fa-lg" aria-hidden="true"></i></a>
      </p></center>
    </nav>
    <p><center><a href="https://github.com/poole/hyde"><i>Powered by Hyde</i></a></center></p>
    <p>
  </div>
</div>


    <div class="content container">
      <div class="post">
  <h1 class="post-title">novel extremophiles: cladogram construction</h1>
  <h2 id="background">Background</h2>

<p>My first introduction to bioinformatics was during my undergraduate research thesis, and the experiences I had inspired me to pursue more computational work for my graduate studies. For my thesis I used traditional laboratory and genomic techniques to characterize a novel bacteria isolate.</p>

<p>This bacterium, referred to as Eh-2, was collected from the intersection of a hot spring and an alkaline lake in Ethiopia. While in the lab I was conducting growth rate studies, I also was interested in it’s 16S rRNA phylogeny. Based on 75% coverage of the gene and a BLAST query, Eh-2 was only 95% similar to one other organism in culture.</p>

<p>Constructing the 16S rRNA phylogenetic tree was an interesting experience for me as a junior scientist. I was mentored by a post-doc in my lab and learned that there wasn’t a single, standard way to create a tree, and that the process took a bit of careful pruning, preparation, and multiple programs to create.</p>

<h2 id="methods">Methods</h2>

<p>This tree was generated through <em>several</em> steps:</p>

<ul>
  <li>ARB was used to locate both similar as well as more “rooted” organisms (using the SILVA database)</li>
  <li>16S rRNA sequences of these organisms were gathered from the NCBI Database</li>
  <li>RAxML was used to construct the newick-formatted tree</li>
  <li>MEGA was used to image and design the tree</li>
  <li>Additional formatting of species names was done using Inkscape</li>
</ul>

<h2 id="results">Results</h2>

<p><img src="/public/images_portfolio/finaltree_forpaper.png" alt="16s rRNA phylogenetic tree" width="100%" /></p>

<p>The above cladogram was generated with my previously mentioned methods. It was included in my undergraduate honors thesis, and included in presentations for both the Oregon NASA Consortium and the Portland State University McNair Scholars Symposium.</p>

<h2 id="future">Future</h2>

<p>I really enjoyed learning about the well-crafted art of phylogenetic tree construction, and have gone on to learn from other mentors that this is an often opinionated field of work. In the future I would use RAxML, but would probably seek out more up-to-date, open-source tools for the rest of the pipeline.</p>

</div>

    </div>

  </body>
</html>
