<!DOCTYPE html>
<html lang="en-us">

  <head>
  <link href="http://gmpg.org/xfn/11" rel="profile">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta http-equiv="content-type" content="text/html; charset=utf-8">

  <!-- Enable responsiveness on mobile devices-->
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">

  <title>
    
      intro to bioinformatics: 16S rRNA microbiome profiling &middot; jmicrobe
    
  </title>

  <!-- CSS -->
  <link rel="stylesheet" href="/public/css/poole.css">
  <link rel="stylesheet" href="/public/css/syntax.css">
  <link rel="stylesheet" href="/public/css/hyde.css">
  <link href='https://fonts.googleapis.com/css?family=Pompiere' rel='stylesheet' type='text/css'>

  <!-- Icons -->
  <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/public/apple-touch-icon-144-precomposed.png">
                                 <link rel="shortcut icon" href="/public/favicon.ico">

  <script src="https://use.fontawesome.com/b4fd5cd984.js"></script>


  <!-- RSS -->
  <link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml">
</head>


  <body>

    <div class="sidebar">
  <div class="container sidebar-sticky">
    <div class="sidebar-about">
      <h1>
        <center><img src="/public/wbunny.png" style="width: 45%;"></center>
        <a href="/">
          jmicrobe
        </a>
      </h1>
      <p class="lead">Microbial ecologist, bioinformatician, mad scientist</p>
    </div>

    <nav class="sidebar-nav">

      

      
      
        
          
        
      
        
          
            <a class="sidebar-nav-item" href="/about/">about me</a>
          
        
      
        
      
        
          
            <a class="sidebar-nav-item" href="/cv/">curriculum vitae</a>
          
        
      
        
          
        
      
        
          
            <a class="sidebar-nav-item" href="/portfolio/">portfolio</a>
          
        
      
        
          
            <a class="sidebar-nav-item" href="/posts/">posts</a>
          
        
      
        
          
            <a class="sidebar-nav-item" href="/tutorials/">tutorials</a>
          
        
      
        
      
      <p><center>
        <a href="https://github.com/jmicrobe"><i class="fa fa-github fa-lg" aria-hidden="true"></i></a>
        <a href="https://www.linkedin.com/in/hardwickejessica"><i class="fa fa-linkedin-square fa-lg" aria-hidden="true"></i></a>
        <a href="https://twitter.com/jessica_microbe"><i class="fa fa-twitter-square fa-lg" aria-hidden="true"></i></a>
        <a href="mailto:jessica@jmicrobe.me"><i class="fa fa-envelope fa-lg" aria-hidden="true"></i></a>
      </p></center>
    </nav>
    <p><center><a href="https://github.com/poole/hyde"><i>Powered by Hyde</i></a></center></p>
    <p>
  </div>
</div>


    <div class="content container">
      <div class="post">
  <h1 class="post-title">intro to bioinformatics: 16S rRNA microbiome profiling</h1>
  <h3 id="view-the-workshop-code-on-github"><a href="https://github.com/jmicrobe/BI331-taxonomy">View the workshop code on Github</a></h3>

<h2 id="objective">Objective</h2>
<p>Modify and update an existing workshop on introductory bioinformatics &amp; microbial
ecology to include more current tools and skills in the field.</p>

<h2 id="background">Background</h2>
<p>The workshop was originally designed as a two-hour class presented to undergraduate
microbiology students. It included some background information on microbial ecology,
and utilized data from <a href="http://www.pnas.org/content/107/33/14691">Filippo et al. (2010)</a> in order to compare the taxonomic differences
between microbiota in the guts of children on a European diet versus children from Africa.</p>

<p>The original workshop utilized the online classifier tool provided by the
Ribosomal Database Project. After data was generated online, students used
Microsoft Excel to clean, calculate and visualize their data.</p>

<h2 id="methods">Methods</h2>

<p>The original background information and data was kept, but the data processing steps
were reworked to focus on command line and programming-based tools. Students used
command line to clone the class files from Github, preview fasta files and count
the number of fasta entries in a file. They then used the RDP classifier via command
line to generate taxonomic and comparative tables that were then visualized using
RStudio.</p>

<h2 id="results">Results</h2>
<p>Below are the figures generated by students:</p>

<p><img src="/public/images_portfolio/abundance_stacked_barplot.png" alt="" title="Microbiome Composition: Burkina Faso vs Europe" /></p>

<p><img src="/public/images_portfolio/significance_clustered_bar.png" alt="" title="Top Significantly Different Phyla: Burkina Faso vs Europe" /></p>

<p>The course was presented to two lab groups, and all students were able to complete the assignment in time. Overall it was very well received. I believe one reason why the course worked so well was that the laptops provided to the students for the exercise were all a unix-based operating system, with the necessary programs and R libraries pre-installed.</p>

<h2 id="future">Future</h2>

<p>I provided the course material on Github so that in the future other graduate students can modify and present this same workshop. In the future I think some tweaks to the format would be ideal, including the use of a <a href="http://jupyter.org/">jupyter notebook</a>, as well as a webpage presenting the material.</p>

</div>

    </div>

  </body>
</html>
